### DISCUSSION TITLE
How can I connect to a Supabase database from IntelliJ IDEA's database tool?

### DISCUSSION THREAD

**Discussion by TheOnlyTails** [supabase/supabase#6418] [Questions] [answered]
I've searched the docs and googled, but found nothing. Inserting the HTTP endpoint doesn't seem to work.

**Comment by TheOnlyTails** [comment, marked as answer]
I've found a solution - I needed to use the JDBC direct connection string. Working perfectly now!

**Comment by bradparks** [comment]
Go here and it's detailed on this page
https://supabase.com/docs/guides/database/connecting-to-postgres#direct-connections

**Comment by limdingwen** [comment]
To any future readers, please remember to delete your user and password from the URL! That gets committed into git. You can specify it in the IntelliJ UI as shown below, and it'll work too.

And yes, use JDBC, not URI.

<img width="796" alt="Screenshot 2024-08-07 at 5 42 18 PM" src="https://github.com/user-attachments/assets/eb830479-427a-42e3-9791-bc6c40ef64a7">

**Comment by rapar8** [comment]
For me direct connection is not working as it does not support IPV4 or need to purchase add on
Go with Transaction pooler
<img width="644" alt="Screenshot 2025-06-13 at 10 36 48 PM" src="https://github.com/user-attachments/assets/e1a32667-86f6-4fb8-8e84-b7bb8ce9c2e1" />

**Comment by piavgh** [comment]
For anyone who wasted an hour like me:
https://supabase.com/docs/guides/database/beekeeper-studio (DataGrip or Jetbrains IDEs in general are the same)

Remember the user is: `postgres.<your_project_id>`, not just `postgres`

**Comment by StacySiz** [comment]
For me user name was just postgres + used Session pooler option
